__tests__/app.test.js

```diff
@@ -1,69 +1,221 @@
-                    author: expect.any(String),
-                    body: expect.any(String),
-                    created_at: expect.any(String),
-                    votes: expect.any(Number),
-                    article_img_url:
-                    expect.any(String),
-                })
-            })
-        })
-    })
-});
\ No newline at end of file
+            expect(response.body.msg).toBe("No comments yet");
+        });
+});
+test("POST:201 posting a new comment", () => {
+    const newComment = {
+        author: "butter_bridge",
+        comment: "did  you know that peppa pig is actually vegetarian?",
+        article_id: 1,
+    };
+    return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.author).toBe(newComment.author);
+            expect(response.body.comment.body).toBe(newComment.comment);
+            expect(response.body.comment.article_id).toBe(
+                newComment.article_id
+            );
+        });
+});
+test("POST: 201 ignore unnecessary properties", () => {
+    const newComment = {
+        author: "butter_bridge",
+        comment: "did  you know that peppa pig is actually vegetarian?",
+        article_id: 1,
+        likesVeggies: true,
+    };
+    return request(app)
+        .post("/api/articles/1/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+            expect(response.body.comment.author).toBe(newComment.author);
+            expect(response.body.comment.body).toBe(newComment.comment);
+            expect(response.body.comment.article_id).toBe(
+                newComment.article_id
+            );
+        });
+});
+test("POST: 404 Username not found", () => {
+    const newComment = {
+        author: "peppapig",
+        comment: "winner winner chicken dinner",
+    };
+    return request(app)
+        .post("/api/articles/2/comments")
+        .send(newComment)
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe("Object not found");
+        });
+});
+test('POST 400, invalid ID, e.g. string of "not-an-id"', () => {
+    const newComment = {
+        author: "peppapig",
+        comment: "winner winner chicken dinner",
+    };
+    return request(app)
+        .post("/api/articles/a/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+            expect(response.body.msg).toBe("Invalid article id");
+        });
+});
+test("POST: 404, non existent article ID", () => {
+    const newComment = {
+        author: "tickle122",
+        comment: "winner winner chicken dinner",
+    };
+    return request(app)
+        .post("/api/articles/999/comments")
+        .send(newComment)
+        .expect(404)
+        .then((response) => {
+            expect(response.body.msg).toBe("Object not found");
+        });
+});
```

Comment:
I would also like to see one more test for Status 400, missing required field(s), e.g. no username or body properties